### PR TITLE
Use pkg/sql/codegen/attribute in XGBoost TrainIR

### LINKED ISSUE #1002: Use pkg/sql/codegen/attribute in XGBoost TrainIR
Part of #805

### PR DESCRIPTION
**Pull request by tonyyang-svail**
It depends on #1004.

Fix #1002.

### CODE REVIEW

**Review by typhoonzero — COMMENTED** [review]
merge https://github.com/sql-machine-learning/sqlflow/pull/1004#pullrequestreview-301623506 first?

**Comment by tonyyang-svail** [comment]
> merge #1004 (review) first?

Done.

File: pkg/sql/codegen/xgboost/codegen.go [hunk 1]
```diff
@@ -17,42 +17,33 @@ import (
...
-		}
-	},
-	"objective": func(x interface{}) error {
-		if _, ok := x.(string); !ok {
-			return fmt.Errorf("objective should be of type string, received %T", x)
-		}
-		return nil
-	},
+func newFloat32(f float32) *float32 {
+	return &f
```
Comment (wangkuiyi) [inline pkg/sql/codegen/xgboost/codegen.go:27]: I am not sure if returning the address of a parameter is to *new* a float value. Maybe 

```go
return &new(float32)
```

**Review by wangkuiyi — APPROVED** [review]

File: pkg/sql/codegen/xgboost/codegen.go (hunk 1 above)
Comment (tonyyang-svail) [reply pkg/sql/codegen/xgboost/codegen.go:27]: Go supports returning address of stack variables: https://stackoverflow.com/questions/52996452/returning-pointer-from-a-local-variable-in-function

**Pull request merged**